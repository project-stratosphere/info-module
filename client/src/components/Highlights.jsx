import PropTypes from 'prop-types';
import React from 'react';
import Highlight from './Highlight';
import {
  Wrapper,
  Title,
  HighlightsWrapper,
} from './styles/Highlights.styles';

const Highlights = props => (
  <Wrapper>
    <Title>
      Home Highlights
    </Title>
    <HighlightsWrapper>
      {
        props.highlights.map((highlight) => (
          <Highlight
            key={highlight.head}
            highlight={highlight}
          />
        ))
      }
    </HighlightsWrapper>
  </Wrapper>
);

Highlights.propTypes = {
  highlights: PropTypes.arrayOf(PropTypes.shape({
    head: PropTypes.string,
    body: PropTypes.string,
  })),
};

Highlights.defaultProps = {
  highlights: [
    {
      head: 'String',
      body: 'String',
    },
  ],
};

export default Highlights;
