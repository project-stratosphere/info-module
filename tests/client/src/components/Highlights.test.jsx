import React from 'react';
import Enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Highlights from '../../../../client/src/components/Highlights';

Enzyme.configure({ adapter: new Adapter() });

const highlights = [
  {
    head: 'Mock Head',
    body: 'Mock Body',
  },
];

describe('Highlights', () => {
  it('should render correctly', () => {
    const output = Enzyme.shallow(<Highlights
      highlights={highlights}
    />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
