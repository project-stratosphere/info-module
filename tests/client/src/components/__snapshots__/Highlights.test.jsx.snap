// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlights should render correctly 1`] = `
<styled.div>
  <styled.div>
    Home Highlights
  </styled.div>
  <styled.div>
    <Highlight
      highlight={
        Object {
          "body": "Mock Body",
          "head": "Mock Head",
        }
      }
      key="Mock Head"
    />
  </styled.div>
</styled.div>
`;
